inference-engine/samples/benchmark_app/utils.hpp

```diff
@@ -70,87 +83,151 @@ std::map<std::string, std::string> parseInputParameters(const std::string parame
+                       ->second.size();
+        if (min_size != max_size) {
+            throw std::logic_error(
+                "Shapes number for every input should be either 1 or should be equal to shapes number of other inputs");
         }
-        // Update shape with batch if needed
-        if (batch_size != 0) {
-            std::size_t batch_index = info.layout.find("N");
-            if ((batch_index != std::string::npos) && (info.shape.at(batch_index) != batch_size)) {
-                info.shape[batch_index] = batch_size;
+    }
+
+    reshape_required = false;
+
+    std::vector<benchmark_app::InputsInfo> info_maps;
+
+    for (size_t i = 0; i < min_size; ++i) {
+        benchmark_app::InputsInfo info_map;
+        for (auto& item : input_info) {
+            benchmark_app::InputInfo info;
+            auto name = item.first;
+            auto descriptor = item.second->getTensorDesc();
+            // Precision
+            info.precision = descriptor.getPrecision();
+            // Partial Shape
+            if (shape_map.count(name)) {
+                std::vector<ngraph::Dimension> parsed_shape;
+                if (shape_map.at(name).size() > 1) {
+                    throw std::logic_error(
+                        "shape command line parameter doesn't support multiple shapes for one input.");
+                }
+                info.partialShape = parsePartialShape(shape_map.at(name)[0]);
                 reshape_required = true;
+            } else {
+                info.partialShape = item.second->getPartialShape();
             }
-        }
-        info_map[name] = info;
-    }
 
-    // Update scale and mean
-    std::map<std::string, std::vector<float>> scale_map = parseScaleOrMean(scale_string, info_map);
-    std::map<std::string, std::vector<float>> mean_map = parseScaleOrMean(mean_string, info_map);
+            if (info.partialShape.is_dynamic() && info.isImage()) {
+                throw std::logic_error(
+                    "benchmark_app supports only binary and random data as input for dynamic models at this moment.");
+            }
 
-    for (auto& item : info_map) {
-        if (item.second.isImage()) {
-            item.second.scale.assign({1, 1, 1});
-            item.second.mean.assign({0, 0, 0});
+            // Tensor Shape
+            if (info.partialShape.is_dynamic() && tensors_shape_map.count(name)) {
+                info.tensorShape = parseTensorShape(tensors_shape_map.at(name)[i % tensors_shape_map.at(name).size()]);
+            } else if (info.partialShape.is_static()) {
+                info.tensorShape = info.partialShape.get_shape();
+            } else if (!tensors_shape_map.empty()) {
+                throw std::logic_error("Wrong input name in tensor_shape command line parameter.");
+            } else {
+                throw std::logic_error(
+                    "tensor_shape command line parameter should be set in case of network dynamic shape.");
```

Comment:
Add a name as hint to make it easier for the user to specify new parameter

Reply:
Done https://github.com/openvinotoolkit/openvino/pull/7788/commits/9d7cfb7e3d2e327ba2fa47614bad411217731a74